sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -892,10 +892,14 @@ def reset(self):
             A boolean, True if successful, False if not
         """
         if self.reset_module(True):
+            # minimum waiting time for the TWI to be functional again
+            time.sleep(2)
+            # buffer time
             for retries in range(5):
-                time.sleep(1)
-                if self.get_module_state() != 'Unknown':
+                state = self.get_module_state()
+                if state in ['ModuleReady']:
```

Comment:
module won't transition to 'ModuleReady' state from 'ModuleLowPwr' unless LowPwrS is FALSE so this check will fail on those cases where LowPwrS is TRUE. so we should expect the module to be in one of the steady state i.e either ModuleReady or ModuleLowPwr

Reply:
The trick is, the reset_module() always clear LowPwrS, but yes, I could improve the reset() accordingly